src/main.cpp

```diff
@@ -166,34 +196,77 @@ void loop()
   char tempString[10];
   dtostrf(temperature, 1, 2, tempString);
 
-  Serial.print("Temperature: ");
-  Serial.print(tempString);
-  Serial.println(" °C");
+  // Serial.print("Temperature: ");
+  // Serial.print(tempString);
+  // Serial.println(" °C");
 
   // Send temperature over BLE
   pTempCharacteristic->setValue(tempString);
-  pTempCharacteristic->notify();
+  // Don't need to notify so often
+  // pTempCharacteristic->notify();
+
+  // Get scale reading
+  float weight = scale.get_units(1);
+
+  // Add reading to the moving average buffer
+  totalWeight -= weightReadings[readingIndex];
+  weightReadings[readingIndex] = weight;
+  totalWeight += weight;
+  readingIndex = (readingIndex + 1) % windowSize;
+
+  // Calculate the moving average
+  averageWeight = totalWeight / windowSize;
+
+  // Convert weight to a string
+  char weightString[20];
+  sprintf(weightString, "%f", weight);
+
+  Serial.print("Weight: ");
+  Serial.print(weightString);
+  Serial.println(" oz");
+  
+  sprintf(weightString, "%f", averageWeight);
+  Serial.print("Average Weight: ");
+  Serial.print(weightString);
+  Serial.println(" oz");
+
+  unsigned long currentTimestamp = millis();
+  if (currentTimestamp - lastUpdateTimestamp >= updateInterval) {
+    pastAverages[pastAveragesIndex] = averageWeight;
+    pastAveragesIndex = (pastAveragesIndex + 1) % pastAveragesSize;
+    lastUpdateTimestamp = currentTimestamp;
+  }
 
-  if (scale.wait_ready_timeout(1000))
```

Comment:
Chris do we need this?

Reply:
if electrical connection is shit it could block so the temp reading won't get sent